<?xml version="1.0" encoding="UTF-8"?>

<launch>

	<arg name="rolename" default="stanford_ego" />


	<!-- bridge -->
	<include file="$(find carla_ros_bridge)/launch/carla_ros_bridge.launch">
		<arg name="host" value="alex" />
    <arg name="ego_vehicle_name" value="$(arg rolename)" />
	</include>


	<!-- shared by all the cars  -->
	<node name="car_filter" pkg="carla_circle" type="car_filter.py" output="screen" />

	<include file="$(find carla_circle)/launch/car_simulation.launch">
		<arg name="rolename" value="$(arg rolename)" />
	</include>

	<!-- visualization -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find carla_ros_bridge)/config/carla_default_rviz.cfg.rviz"/>

	<!-- game planner -->
  <include file="$(find uavgame)/launch/main_mpc_gurobi.launch" >
		<arg name="role_name" value="$(arg rolename)" />
	</include>

</launch>
